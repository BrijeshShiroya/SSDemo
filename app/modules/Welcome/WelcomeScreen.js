import React, { useCallback, useEffect, useState } from 'react';
import { Image, Platform, View } from 'react-native';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import {
  AppBackground,
  CustomButton,
  CustomTextInput,
  Greets
} from '../../components';
import { Strings } from '../../constants';
import WelcomeActions from '../../redux/WelcomeRedux';
import { Colors, Icons } from '../../theme';
import styles from './styles/WelcomeScreenStyle';
import Info from '../../services/DeviceTypeChecker';
import { showToast } from '../../services/Utils';

const WelcomScreen = ({ navigation }) => {
  const [userName, setUserName] = useState('');
  const [emulator, setEmulator] = useState(null);
  const dispatch = useDispatch();

  useEffect(() => {
    Info.isEmulator().then((isEmulator) => setEmulator(isEmulator));
  }, [setEmulator]);
  const renderUserName = () => {
    return (
      <View style={styles.inputContainer}>
        <Image source={Icons.user} style={styles.left} />
        <CustomTextInput
          value={userName}
          placeholder={Strings.userNamePlaceholder}
          placeholderTextColor={Colors.txtBorder}
          style={styles.input}
          onChangeText={(text) => setUserName(text)}
        />
      </View>
    );
  };

  const onSubmitPress = useCallback(() => {
    if (userName === '') {
      showToast('Please enter username');
    } else {
      dispatch(WelcomeActions.welcomeRequest({ username: userName }));
      navigation.navigate('HomeScreen');
    }
  }, [userName, dispatch, navigation]);

  const renderFooter = () => {
    return (
      <CustomButton
        title={Strings.btnSubmit}
        style={styles.btnLogin}
        onPress={onSubmitPress}
      />
    );
  };

  const renderIsEmulator = () => {
    const title = emulator
      ? `You are running app in ${
          Platform.OS === 'ios' ? 'simulator' : 'emulator'
        }`
      : 'You are running app in real device';
    return <Greets title={title} style={styles.greets} />;
  };

  return (
    <AppBackground>
      <View style={styles.container}>
        {renderIsEmulator()}
        {renderUserName()}
        {renderFooter()}
      </View>
    </AppBackground>
  );
};
WelcomScreen.propTypes = {
  navigation: PropTypes.object
};

export default WelcomScreen;
