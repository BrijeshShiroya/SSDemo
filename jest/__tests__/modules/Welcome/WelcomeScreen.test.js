import 'react-native';
import React from 'react';
import WelcomeScreen from '../../../../app/modules/Welcome/WelcomeScreen';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
const initialState = {
  welcome: {
    user: '',
    fetching: false,
    error: null
  }
};
const mockStore = configureStore();
const store = mockStore(initialState);
// Note: test renderer must be required after react-native.

const wrapper = shallow(
  <Provider store={store}>
    <WelcomeScreen />
  </Provider>
);
describe('WelcomeScreen', () => {
  it('WelcomeScreen renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Change username value', () => {
    const setUserName = jest.fn();
    const handleClick = jest.spyOn(React, 'useState');
    handleClick.mockImplementation((userName) => [userName, setUserName]);
    expect(setUserName).toBeTruthy();
  });
});
