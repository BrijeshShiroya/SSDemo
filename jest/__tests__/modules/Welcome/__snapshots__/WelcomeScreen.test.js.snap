// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen WelcomeScreen renders correctly 1`] = `ShallowWrapper {}`;
